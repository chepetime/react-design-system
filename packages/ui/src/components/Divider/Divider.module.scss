@use "./../../styles" as styles;

.Divider {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  width: 100%;
  height: 1px;
  margin: 0 8px;
  background-color: #ccc;
  border: none;
}
