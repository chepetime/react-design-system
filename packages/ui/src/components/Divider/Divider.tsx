import React from "react";
import classnames from "classnames";
import css from "./Divider.module.scss";

export interface DividerProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Divider = React.forwardRef<HTMLDivElement, DividerProps>(
  (props: DividerProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Divider)} ref={ref}>
        <p>Divider component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Divider.displayName = "Divider";
