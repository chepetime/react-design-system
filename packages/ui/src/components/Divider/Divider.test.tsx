import React from "react";import { render } from "@testing-library/react";
import { Divider } from "./Divider";

describe("Divider", () => {
  test("renders the Divider component", () => {
    render(<Divider />);
  });
});
