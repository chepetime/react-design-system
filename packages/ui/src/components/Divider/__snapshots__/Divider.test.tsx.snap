// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider matches snapshot 1`] = `
<div>
  <div
    class="Divider"
  >
    <p>
      Divider component working!
    </p>
    <div />
  </div>
</div>
`;
